ui-participant/src/login/RedirectFromOAuth.tsx

```diff
@@ -91,9 +93,31 @@ export const RedirectFromOAuth = () => {
 
             // Enroll in the study if not already enrolled in any other study
             if (defaultEnrollStudy && !loginResult.enrollees.length) {
-              const hubUpdate = await enrollCurrentUserInStudy(defaultEnrollStudy.shortcode,
-                defaultEnrollStudy.name, preEnrollResponseId, refreshLoginState, i18n)
-              navigate('/hub', { replace: true, state: hubUpdate })
+              const hubResponse = await enrollCurrentUserInStudy(
+                defaultEnrollStudy.shortcode, preEnrollResponseId, refreshLoginState)
+
+              const sortedActiveConsentTasks = getSortedActiveTasks(hubResponse.enrollee.participantTasks, 'CONSENT')
+              const nextConsentTask = getNextTask(hubResponse.enrollee, sortedActiveConsentTasks)
+
+              if (nextConsentTask) {
+                //if there's a pending consent, navigate directly to the first one. note that we don't
+                //show the "Welcome to the study" banner in this case
+                const consentTaskPath = getTaskPath(
+                  nextConsentTask, hubResponse.enrollee.shortcode, defaultEnrollStudy.shortcode)
+                navigate(`../hub/${consentTaskPath}`, { replace: true })
+              } else {
+                const hubUpdate: HubUpdate = {
+                  message: {
+                    title: i18n('hubUpdateWelcomeToStudyTitle', {
+                      substitutions: { studyName: defaultEnrollStudy.name }
+                    }),
+                    detail: i18n('hubUpdateWelcomeToStudyDetail'),
+                    type: 'INFO'
+                  }
+                }
```

Comment:
this also feels like it might go in a helper somewhere, but I'm less sure about this one